frontend/src/app/socketio.ts

```diff
@@ -22,20 +22,16 @@ import {
     frontendToBackendParameters,
 } from './parameterTranslation';
 
-export interface SocketioMiddlewareConfig {
-    host: string;
-    port: number;
-}
-
 export interface SocketIOResponse {
     status: 'OK' | 'ERROR';
     message?: string;
     data?: any;
 }
 
-export const socketioMiddleware = (config: SocketioMiddlewareConfig) => {
-    const { host, port } = config;
-    const socketio = io(`http://${host}:${port}`);
+export const socketioMiddleware = () => {
+    const { hostname, port } = new URL(window.location.href);
+
+    const socketio = io(`http://${hostname}:${parseInt(port)}`);
```

Comment:
This will not work. Because in dev mode the port will be 5173 and won't connect to the server which is on 9090.

Just wondering, how are you doing the development of this on your end? You don't have any proxies setup. So your images won't load. Are you actually building each time you want to test?